homeassistant/components/recorder/pool.py

```diff
@@ -31,37 +29,52 @@
 )
 
 
-class RecorderPool(SingletonThreadPool, NullPool):  # type: ignore[misc]
+class RecorderPool(SingletonThreadPool, NullPool):
     """A hybrid of NullPool and SingletonThreadPool.
 
     When called from the creating thread or db executor acts like SingletonThreadPool
     When called from any other thread, acts like NullPool
     """
 
     def __init__(  # pylint: disable=super-init-not-called
-        self, *args: Any, **kw: Any
+        self,
+        creator: Any,
+        recorder_and_worker_thread_ids: set[int] | None = None,
+        **kw: Any,
     ) -> None:
         """Create the pool."""
         kw["pool_size"] = POOL_SIZE
-        SingletonThreadPool.__init__(self, *args, **kw)
-
-    @property
-    def recorder_or_dbworker(self) -> bool:
-        """Check if the thread is a recorder or dbworker thread."""
-        thread_name = threading.current_thread().name
-        return bool(
-            thread_name == "Recorder" or thread_name.startswith(DB_WORKER_PREFIX)
+        assert (
+            recorder_and_worker_thread_ids is not None
+        ), "recorder_and_worker_thread_ids is required"
+        self.recorder_and_worker_thread_ids = recorder_and_worker_thread_ids
+        SingletonThreadPool.__init__(self, creator, **kw)
+
+    def recreate(self) -> "RecorderPool":
+        """Recreate the pool."""
+        self.logger.info("Pool recreating")
+        return self.__class__(
```

Comment:
Copied from `SingletonThreadPool`. Sadly there doesn't seem to be away to avoid writing this out